### PR TITLE
Drop Support for python 3.6

### LINKED ISSUE #360: CI for python 3.6  broken [BUG]
**What happened**:
The CI for [python 3.6](https://github.com/xgcm/xgcm/runs/3161900333?check_suite_focus=true) is broken. I suggest we drop support for these versions, which are not supported by xarray anymore since [0.17.0](http://xarray.pydata.org/en/stable/whats-new.html?highlight=3.6#id53)

### PR DESCRIPTION
**Pull request by jbusecke**
- [x] Closes #360 
 - [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by jbusecke** [comment]
Is there anything else I need to do to drop the version support?

**Review by dcherian — APPROVED** [review]
I think this is all that's neede

**Review by andersy005 — APPROVED** [review]
You could also update the minimum required version to `v0.17` (which is the version that dropped python 3.6 in xarray). 



https://github.com/xgcm/xgcm/blob/6e68f71caad8f57e2776d4dc7b4aaa3eefb1875f/setup.py#L10

File: setup.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
     long_description = f.read()
 
 install_requires = [
-    "xarray>=0.14.1",
+    "xarray>=0.17.0",
     "dask",
     "numpy",
     "future",
```
Comment (andersy005) [inline setup.py:13]: Is this still needed? I thought it was only needed for applications that wanted to support Python 2 & Python 3

Comment (jbusecke) [reply setup.py:13]: I think you are right. I just removed it. Lets see if that upsets the CI.

**Pull request merged**